pkg/api/admin/openshiftcluster.go

```diff
@@ -69,6 +70,11 @@ type ClusterProfile struct {
 	ResourceGroupID string `json:"resourceGroupId,omitempty"`
 }
 
+// FeatureProfile represents a feature profile.
+type FeatureProfile struct {
+	GatewayEnabled bool `json:"gatewayEnabled,omitempty" mutable:"true"`
+}
+
```

Comment:
We doing this on cluster side? Does this implies Gateway will require new API?

Reply:
No - this is an internal switch which lets us administratively roll out the gateway architecture to existing clusters on a per-cluster basis.